valhalla/sif/edgelabel.h

```diff
@@ -49,6 +49,7 @@ class EdgeLabel {
             const float dist,
             const TravelMode mode,
             const uint32_t path_distance,
+            const Cost& transition_cost,
```

Comment:
here i refactored edgelabel to move the transition_cost to the base class and then force all the derived classes to set it in the constructor and on calls to `update`